x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -447,38 +482,116 @@ export class AnnotationsTable extends Component {
       });
     }
 
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      width: '60px',
-      name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
-        defaultMessage: 'Actions',
-      }),
-      actions,
-    });
-
     const getRowProps = (item) => {
       return {
         onMouseOver: () => this.onMouseOverRow(item),
         onMouseLeave: () => this.onMouseLeaveRow(),
       };
     };
+    let filterOptions = [];
+    if (this.props.aggregations?.event?.buckets) {
+      const buckets = this.props.aggregations.event.buckets;
+      const foundUser = buckets.findIndex((d) => d.key === ANNOTATION_EVENT_USER) > -1;
+      filterOptions = foundUser
+        ? buckets
+        : [{ key: ANNOTATION_EVENT_USER, doc_count: 0 }, ...buckets];
+    }
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'event',
+        name: 'Event',
+        multiSelect: 'or',
+        options: filterOptions.map((field) => ({
+          value: field.key,
+          name: field.key,
+          view: `${field.key} (${field.doc_count})`,
+        })),
+      },
+    ];
+
+    if (
+      this.props.chartDetails?.entityData?.entities &&
+      this.props.chartDetails?.entityData?.entities.length > 0
+    ) {
+      this.props.chartDetails?.entityData?.entities.forEach((entity) => {
+        if (entity.fieldType === 'partition_field') {
+          columns.push({
+            field: 'partition_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.partitionColumnName', {
+              defaultMessage: 'Partition',
+            }),
+            sortable: true,
+          });
+        }
+
+        if (entity.fieldType === 'over_field') {
+          columns.push({
+            field: 'over_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.overColumnName', {
+              defaultMessage: 'Over',
+            }),
+            sortable: true,
+          });
+        }
+        if (entity.fieldType === 'by_field') {
+          columns.push({
+            field: 'by_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.byColumnName', {
+              defaultMessage: 'By',
+            }),
+            sortable: true,
+          });
+        }
+      });
+
+      filters.push({
+        type: 'is',
+        field: CURRENT_SERIES,
+        name: i18n.translate('xpack.ml.annotationsTable.seriesOnlyFilterName', {
+          defaultMessage: 'Filter to series',
+        }),
+      });
+    }
+    const search = {
+      defaultQuery: queryText,
+      box: {
+        incremental: true,
+        schema: true,
+      },
+      filters: filters,
+    };
+
+    columns.push(
```

Comment:
We also need to show the detector description in the table, as we do for the anomalies. This could be done in a follow-up. You'll just need to get the detector description from the `detector_index` stored for the annotation.

![image](https://user-images.githubusercontent.com/7405507/86463198-f67bc400-bd24-11ea-86ff-5a07158ec8ad.png)

Reply:
Added detector column in SMV here  https://github.com/elastic/kibana/pull/70198/commits/1f2684dbbc91ca4a7d83ab93d8d4cfb79a5180c2
<img width="1787" alt="Screen Shot 2020-07-08 at 5 12 48 PM" src="https://user-images.githubusercontent.com/43350163/86975657-5c5db680-c13e-11ea-9e6c-5d6446c2f066.png">

Reply:
As above, may be best not to show the detector column after all. The user can always see it in the edit flyout.